src/components/include/utils/threads/message_loop_thread.h

```diff
@@ -147,9 +147,8 @@ MessageLoopThread<Q>::MessageLoopThread(const std::string& name,
 
 template <class Q>
 MessageLoopThread<Q>::~MessageLoopThread() {
-  Shutdown();
-  delete thread_delegate_;
   threads::DeleteThread(thread_);
```

Comment:
@SNiukalov `thread_` contains `delegate` . So why did you delete first containing object and only after you delete part of it (its delegate)?

Reply:
We used aggregation in Thread class and we dont have contorol over delegate life cycle, so logically need first to delete container and then delete delegate. we are calling join() -> stop() Inside destuctor of Thread we are using delegate_->exitThreadMain(); Inside the stop()